api/v1alpha1/proxydefaults_types.go

```diff
@@ -0,0 +1,208 @@
+type ProxyDefaultsSpec struct {
+	// Config is an arbitrary map of configuration values used by Connect proxies.
+	// Any values that your proxy allows can be configured globally here.
+	// Supports JSON config values. See https://www.consul.io/docs/connect/proxies/envoy#configuration-formatting
+	Config json.RawMessage `json:"config,omitempty"`
+	// MeshGateway controls the default mesh gateway configuration for this service.
+	MeshGateway MeshGatewayConfig `json:"meshGateway,omitempty"`
+	// Expose controls the default expose path configuration for Envoy.
+	Expose ExposeConfig `json:"expose,omitempty"`
+}
+
+func (in *ProxyDefaults) GetObjectMeta() metav1.ObjectMeta {
+	return in.ObjectMeta
+}
+
+func (in *ProxyDefaults) AddFinalizer(name string) {
+	in.ObjectMeta.Finalizers = append(in.Finalizers(), name)
+}
+
+func (in *ProxyDefaults) RemoveFinalizer(name string) {
+	var newFinalizers []string
+	for _, oldF := range in.Finalizers() {
+		if oldF != name {
+			newFinalizers = append(newFinalizers, oldF)
+		}
+	}
+	in.ObjectMeta.Finalizers = newFinalizers
+
+}
+
+func (in *ProxyDefaults) Finalizers() []string {
+	return in.ObjectMeta.Finalizers
+}
+
+func (in *ProxyDefaults) ConsulKind() string {
+	return capi.ProxyDefaults
+}
+
+func (in *ProxyDefaults) ConsulNamespaced() bool {
+	return false
+}
+
+func (in *ProxyDefaults) KubeKind() string {
+	return ProxyDefaultsKubeKind
+}
+
+func (in *ProxyDefaults) SyncedCondition() (status corev1.ConditionStatus, reason, message string) {
+	cond := in.Status.GetCondition(ConditionSynced)
+	if cond == nil {
+		return corev1.ConditionUnknown, "", ""
+	}
+	return cond.Status, cond.Reason, cond.Message
+}
+
+func (in *ProxyDefaults) SyncedConditionStatus() corev1.ConditionStatus {
+	cond := in.Status.GetCondition(ConditionSynced)
+	if cond == nil {
+		return corev1.ConditionUnknown
+	}
+	return cond.Status
+}
+
+func (in *ProxyDefaults) Name() string {
+	return in.ObjectMeta.Name
+}
+
+func (in *ProxyDefaults) SetSyncedCondition(status corev1.ConditionStatus, reason string, message string) {
+	in.Status.Conditions = Conditions{
+		{
+			Type:               ConditionSynced,
+			Status:             status,
+			LastTransitionTime: metav1.Now(),
+			Reason:             reason,
+			Message:            message,
+		},
+	}
+}
+
+func (in *ProxyDefaults) ToConsul() api.ConfigEntry {
+	consulConfig := in.convertConfig()
+	return &capi.ProxyConfigEntry{
+		Kind:        in.ConsulKind(),
+		Name:        in.Name(),
+		MeshGateway: in.Spec.MeshGateway.toConsul(),
+		Expose:      in.Spec.Expose.toConsul(),
+		Config:      consulConfig,
+	}
+}
+
+func (in *ProxyDefaults) MatchesConsul(candidate api.ConfigEntry) bool {
+	proxyDefCand, ok := candidate.(*capi.ProxyConfigEntry)
+	if !ok {
+		return false
+	}
+	proxyDefCand.Namespace = ""
+	proxyDefCand.CreateIndex = 0
+	proxyDefCand.ModifyIndex = 0
+
+	return reflect.DeepEqual(in.ToConsul().(*capi.ProxyConfigEntry), proxyDefCand)
```

Comment:
```suggestion
	return reflect.DeepEqual(in.ToConsul(), proxyDefCand)
```

I didn't need this for the others.